concordium-node/src/bin/cli.rs

```diff
@@ -198,38 +248,42 @@ async fn main() -> anyhow::Result<()> {
     // Start baking
     consensus.start_baker();
 
-    // Everything is running, so we wait for a signal to shutdown.
-    if shutdown_signal_1.recv().await.is_err() {
-        error!("Shutdown signal handler was dropped unexpectedly. Shutting down.");
+    // Everything is running, so we wait for a signal to shutdown or for an error to
+    // occur.
+    if shutdown_receiver.recv().await.is_err() {
+        error!("Node shutdown channel was dropped unexpectedly. Shutting down.");
     }
 
-    // Message rpc to shutdown first
-    if let Some(task) = rpc_server_task {
-        if shutdown_rpc_sender.send(()).is_err() {
-            error!("Could not stop the RPC server correctly. Forcing shutdown.");
-            task.abort();
-        }
-        // Force the rpc server to shut down in at most 10 seconds.
-        let timeout_duration = std::time::Duration::from_secs(10);
-        match tokio::time::timeout(timeout_duration, task).await {
-            Ok(res) => {
-                if let Err(err) = res {
-                    if err.is_cancelled() {
-                        info!("RPC server was successfully shutdown by force.");
-                    } else if err.is_panic() {
-                        error!("RPC server panicked: {}", err);
+    // Message rpc to shutdown first, if it did not crash.
+    if !rpc_error.load(atomic::Ordering::Acquire) {
```

Comment:
Since the runtime will kill this task when shutting down, should we wait for thread even if it crashed, or do we not care about it once this happens?